v03_pipeline/lib/tasks/update_variant_annotations_table_with_deleted_families.py

```diff
@@ -25,72 +21,33 @@ def __init__(self, *args, **kwargs):
         self.done = False
 
     def requires(self) -> luigi.Task:
-        return HailTableTask(
-            lookup_table_path(
-                self.reference_genome,
-                self.dataset_type,
-            ),
+        return UpdateLookupTableWithDeletedFamiliesTask(
+            dataset_type=self.dataset_type,
+            sample_type=self.sample_type,
+            reference_genome=self.reference_genome,
+            project_guid=self.project_guid,
+            family_guids=self.family_guids,
         )
 
     def complete(self) -> bool:
         if not self.dataset_type.has_lookup_table:
             return True
-        return super().complete() and (
-            # We don't have the concept of families being present or
-            # not present in annotations table, thus we check the "lookup"
-            # table OR a done flag to prevent the task from looping over itself.
-            self.done
-            or hl.eval(
-                hl.bind(
-                    lambda family_guids: (
-                        hl.is_missing(family_guids)  # The project itself is missing
-                        | hl.all(
-                            hl.array(list(self.family_guids)).map(
-                                lambda family_guid: ~hl.set(family_guids).contains(
-                                    family_guid,
-                                ),
-                            ),
-                        )
-                    ),
-                    hl.read_table(self.input().path).globals.project_families.get(
-                        self.project_guid,
-                    ),
-                ),
-            )
-        )
+        # We don't have the concept of families being present or not present
+        #  in annotations table a done flag to prevent the task from looping over itself.
+        return super().complete() and self.done
 
     def update_table(self, ht: hl.Table) -> hl.Table:
         if not self.dataset_type.has_lookup_table:
             return ht
         lookup_ht = hl.read_table(self.input().path)
-        project_i = hl.eval(lookup_ht.globals.project_guids.index(self.project_guid))
```

Comment:
I decided to just remove this optimization (to only annotate changed rows) because the logic got even more confusing with the lookup table potentially changing its row count.